### PR TITLE
fix(config): create jestPlusTSJest group preset

### LINKED ISSUE #10594: Update jest and ts-jest major versions together
**What would you like Renovate to be able to do?**

I would like Renovate to update major versions of jest and ts-jest together. As a recent example, jest v27 introduced breaking changes, so jest v27 and ts-jest v26 don't work together. ts-jest was then updated to v27 to work with jest v27, but this means that it no longer worked with jest v26. Since Renovate updates each package separately, it created 2 PRs that had failing tests, and the only resolution was to manually update both packages at the same time without Renovate's help.

Obviously the jest packages don't get a major update *that* often, but I imagine that there are similar cases where 2 non-monorepo packages pretty much always need to be updated at the same time.

Renovate could still run tests for the first package that was updated, and if it passes then it can go ahead and merge, but if not then it will wait to merge both at the same time.

**Did you already have any implementation ideas?**

Renovate already has support for "monorepo" packages, so I assume the implementation path here would be to create something similar but which is only triggered for major versions as opposed to all versions.

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes:

- Creates new group to bundle Jest and TS-Jest major updates together

## Context:

The idea is to group major Jest and TS-Jest updates into one PR, as things will break if you update them separately.
I don't think all Jest users use TypeScript, so this is a new group preset, instead of fixing up the Jest monorepo preset.

There's also this message on the `ts-jest` repo readme:

> We DO NOT use SemVer for versioning. Though you can think about SemVer when reading our version, except our major number follows the one of Jest. For the versions available, see the tags on this repository.

So we should only group **major** updates, as those are the only ones that are matched to the upstream Jest versioning.

Fixes #10594.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/config/presets/internal/group.ts [hunk 1]
```diff
@@ -572,6 +572,19 @@ const staticGroups = {
       },
     ],
   },
+  JestPlusTSJest: {
```
Comment (HonkingGoose) [inline lib/config/presets/internal/group.ts:575]:
```suggestion
  jestPlusTSJest: {
```

**Review by viceice — DISMISSED** [review]
Not sure if we should enable this by default in base preset. 🤔 Suggestions ?

**Comment by HonkingGoose** [comment]
> Not sure if we should enable this by default in base preset. 🤔 Suggestions ?

Did I put it in the base preset section then? That is not what I want, as I think only a subset of Jest users will use ts-jest. The idea is that people who need this group preset can opt-in manually, while things keep on working as they used to for everybody else.

**Comment by viceice** [comment]
No, it's not added, but I would prefer to add ts-jest to jest monorepo for major updates only.

So it's always grouped by default if it's found. 🤔

**Comment by HonkingGoose** [comment]
> No, it's not added, but I would prefer to add ts-jest to jest monorepo for major updates only.
> 
> So it's always grouped by default if it's found. 🤔

So, if I understand you correctly, you'd like to change

https://github.com/renovatebot/renovate/blob/226ec075c33d92e1999cd515ac74ae660c15af8f/lib/config/presets/internal/monorepo.ts#L102

into:

```
  jest: [
    'https://github.com/facebook/jest',
    'https://github.com/kulshekhar/ts-jest',
  ],
```
???

**Comment by viceice** [comment]
No, i think a new preset for ts-jest only.

Set group name to same like jest monorepo and match major updates only.

File: lib/config/presets/internal/group.ts [hunk 2]
```diff
@@ -572,6 +572,19 @@ const staticGroups = {
       },
     ],
   },
+  jestPlusTSJest: {
```
Comment (HonkingGoose) [inline lib/config/presets/internal/group.ts:575]:
```suggestion
  jest: {
```

File: lib/config/presets/internal/group.ts [hunk 3]
```diff
@@ -572,6 +572,19 @@ const staticGroups = {
...
+    description: 'Group major updates for jest and ts-jest together',
```
Comment (HonkingGoose) [inline lib/config/presets/internal/group.ts:576]:
```suggestion
    description: 'Group all major jest packages together',
```

File: lib/config/presets/internal/group.ts [hunk 4]
```diff
@@ -572,6 +572,19 @@ const staticGroups = {
...
+    packageRules: [
+      {
+        matchSourceUrlPrefixes: [
+          'https://github.com/facebook/jest',
+          'https://github.com/kulshekhar/ts-jest',
+        ],
+        matchUpdateTypes: ['major'],
+        groupName: 'ts-jest and jest',
```
Comment (HonkingGoose) [inline lib/config/presets/internal/group.ts:584]:
```suggestion
        groupName: 'jest monorepo,
```

**Review by HonkingGoose — COMMENTED** [review]
@viceice Is this what you want? I'm not 100% sure what you want me to do, so this is my best-effort guess... 😉

File: lib/config/presets/internal/group.ts (hunk 2 above)
Comment (viceice) [reply lib/config/presets/internal/group.ts:575]: leave as is

File: lib/config/presets/internal/group.ts (hunk 3 above)
Comment (viceice) [reply lib/config/presets/internal/group.ts:576]:
```suggestion
    description: 'Group major updates for ts-jest to jest monorepo',
```

Something like this?

File: lib/config/presets/internal/group.ts [hunk 5]
```diff
@@ -572,6 +572,19 @@ const staticGroups = {
       },
     ],
   },
+  jestPlusTSJest: {
+    description: 'Group major updates for jest and ts-jest together',
+    packageRules: [
+      {
+        matchSourceUrlPrefixes: [
+          'https://github.com/facebook/jest',
```
Comment (viceice) [inline lib/config/presets/internal/group.ts:580]:
```suggestion
```

**Review by viceice — COMMENTED** [review]
Should be enabled by default 🤔

**Comment by viceice** [comment]
image

**Review by viceice — COMMENTED** [review]
LGTM. @rarkins @JamieMagee Should we add this to `config:base` ?

**Comment by rarkins** [comment]
Yes, via group:recommended

**Comment by HonkingGoose** [comment]
> LGTM. ... Should we add this to `config:base` ?

> Yes, via group:recommended


@rarkins I've pushed a new commit, is this what you wanted me to do?

**Review by rarkins — APPROVED** [review]

**Pull request merged**